polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -566,17 +566,30 @@ pub(crate) async fn handle_active_leaves_update<Context>(
 		state.implicit_view.all_allowed_relay_parents().cloned().collect::<Vec<_>>();
 
 	for new_relay_parent in new_relay_parents.iter().cloned() {
+		if state.per_relay_parent.contains_key(&new_relay_parent) {
+			continue
+		}
+
 		let disabled_validators: HashSet<_> =
-			polkadot_node_subsystem_util::request_disabled_validators(
+			match polkadot_node_subsystem_util::request_disabled_validators(
```

Comment:
wouldn't we have the same problem with session index query below?
i would propose to extract individual leaf handling into `handle_active_leaf_update` and keep `?` in there, but in `handle_active_leaves_update` do
```
for leaf in leaves {
  if let Err(err) = handle_active_leaf_update() {
    ...
  }
}
```

Reply:
Good point. I think it can happen for even more stuff down the line, but because we only process once per leaf, it shouldn't fail. 

I would also consider raising the cache size for all runtime APIs used to process/import dispute statements . This would reduce chances that other stuff fails when a node gets hammered by disputes.

Reply:
> I would propose to extract individual leaf handling into handle_active_leaf_update and keep ? in there, but in handle_active_leaves_update

Done.